web/deploy/terraform/staging/main.tf

```diff
@@ -1,102 +1,24 @@
 terraform {
-  required_version = ">= 1.0.0, < 2.0.0"
+  required_version = ">= 1.8.0, < 2.0.0"
 
   required_providers {
     aws = {
       source  = "hashicorp/aws"
       version = "~> 5.0"
     }
   }
-}
-
-provider "aws" {
-  region = "us-east-1"
-}
 
-terraform {
   backend "s3" {
-    bucket         = "osm-terraform-storage"
-    key            = "terraform/staging-state/terraform.tfstate"
-    region         = "us-east-1"
-    dynamodb_table = "terraform-locks"
-  }
-}
-
-
-module "shared_resources" {
-  source = "../modules/shared_resources"
-}
-
-# EC2 Instance
-resource "aws_instance" "staging" {
-  ami                         = module.shared_resources.ami_id
-  instance_type               = var.instance_type
-  subnet_id                   = module.shared_resources.subnet_id
-  key_name                    = "dsst2023"
-  vpc_security_group_ids      = [module.shared_resources.security_group_id]
-  associate_public_ip_address = true
-  root_block_device {
-    volume_size = 30
-    volume_type = "gp2" # General Purpose SSD (can be "gp2", "gp3", "io1", "io2", etc.)
-  }
-
-  tags = {
-    Name = "staging-instance"
-  }
-
-  user_data = <<-EOF
-    #!/bin/bash
-    apt-get update -y
-    apt install -y curl
-    apt-get install -y docker.io
-    curl -SL https://github.com/docker/compose/releases/download/v2.29.1/docker-compose-linux-x86_64 -o /usr/local/bin/docker-compose
-    chmod a+x /usr/local/bin/docker-compose
-    systemctl restart sshd
-    systemctl start docker
-    systemctl enable docker
-              EOF
-}
-
-resource "aws_eip" "staging" {
-  domain = "vpc"
-
-  tags = {
-    Name = "staging-elastic-ip"
+    bucket         = "${var.state_bucket_name}-${var.environment}"
+    key            = var.state_backend_key
+    region         = var.state_storage_region
+    dynamodb_table = "${var.state_table_name}-${var.environment}"
+    encrypt        = true
   }
 }
 
-resource "aws_eip_association" "staging" {
-  instance_id   = aws_instance.staging.id
-  allocation_id = aws_eip.staging.id
-}
-
-output "vpc_id" {
-  value = module.shared_resources.vpc_id
-}
-output "internet_gateway_id" {
-  value = module.shared_resources.internet_gateway_id
-}
-output "route_table_id" {
-  value = module.shared_resources.route_table_id
-}
-output "network_acl_id" {
-  value = module.shared_resources.aws_network_acl_id
-}
-output "security_group_id" {
-  value = module.shared_resources.security_group_id
-}
-output "subnet_id" {
-  value = module.shared_resources.subnet_id
-}
-
-output "instance_id" {
-  value = aws_instance.staging.id
-}
-
-output "public_dns" {
-  value = aws_eip.staging.public_dns
-}
-
-output "public_ip" {
-  value = aws_eip.staging.public_ip
+module "ec2" {
+  source        = "../modules/ec2/"
```

Comment:
This will need to be changed to a github URL with a reference to a tag after this PR is merged